lib/posix/options/pipe.c

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) 2024, Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/sys/fdtable.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/zvfs/eventfd.h>
+#include <zephyr/sys/bitarray.h>
+#include <zephyr/sys/fdtable.h>
+
+struct pipe_obj {
+	struct k_poll_signal read_sig;
+	struct k_poll_signal write_sig;
+	struct k_spinlock lock;
+	/* data buffer is tiny! -> room for improvement */
+	uintptr_t data;
```

Comment:
Sure looks like this could just be a byte array? Why the complexity of storing bytes in a uintptr_t?

Reply:
Mainly because it's small-ish and doesn't introduce another maximum without a standard value. The standard buffer size is supposed to be 512 bytes as well, but there is no standard maximum number of pipes.

But then there is also the classic issue with a byte array about read / write pointers. I'm almost thinking it would be better to use a ring buffer. There are a number of tweaks that can be made, but really just having the interface there is what is important.

Reply:
I guess I'd rather not have the interface here if there isn't a "real" implementation underneath. It's a nice demonstration of the interface ideas, but it looks like a trap for someone who actually needs pipes.

Reply:
@keith-packard - you drive a hard bargain, sir! Let me see what I can whip up

Reply:
@keith-packard - Fixed!